k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -2005,4 +1611,109 @@ public void messageUidChanged(Account account, String folder, String oldUid, Str
         }
     }
 
+    AttachmentMvpView attachmentMvpView = new AttachmentMvpView() {
+        private HashMap<Attachment,View> attachmentViews = new HashMap<>();
```

Comment:
Keeping a map of attachments to views here, to allow reliable update of views. This is transient state which will be lost during configuration changes, it's the presenter's job to keep and restore the state.